cpu/cc26x2_cc13x2/setup.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2020 Locha Inc
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cc26x2_cc13x2
+ * @{
+ *
+ * @file
+ * @brief       CC26x2/CC13x2 Device setup functions
+ *
+ * @author      Jean Pierre Dudey <jeandudey@hotmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+
+static void _trim_device(uint32_t rev);
+
+static void _set_boot_det(void) {
+    const uint32_t clr_m = AON_PMCTL_RESETCTL_BOOT_DET_1_CLR_m |
+                           AON_PMCTL_RESETCTL_BOOT_DET_0_CLR_m |
+                           AON_PMCTL_RESETCTL_BOOT_DET_1_SET_m |
+                           AON_PMCTL_RESETCTL_BOOT_DET_0_SET_m |
+                           AON_PMCTL_RESETCTL_MCU_WARM_RESET_m;
+    const uint32_t boot_det_m = AON_PMCTL_RESETCTL_BOOT_DET_1_m |
+                                AON_PMCTL_RESETCTL_BOOT_DET_0_m;
+    const uint32_t boot_det_s = AON_PMCTL_RESETCTL_BOOT_DET_0_s;
+    reg32_t reg = AON_PMCTL->RESETCTL;
+
+    /**
+     * Set BOOT_DET bits to 3 if already found to be 1
+     * Note: The BOOT_DET_x_CLR/SET bits must be manually cleared
+     */
+    if (((reg & boot_det_m) >> boot_det_s) == 1) {
+        reg &= ~clr_m;
+        AON_PMCTL->RESETCTL = reg | AON_PMCTL_RESETCTL_BOOT_DET_1_SET_m;
+        AON_PMCTL->RESETCTL = reg;
+    }
+}
+
+void setup_trim_device(void)
+{
+    /* Get layout revision of the factory configuration area. Handle undefined
+     * revision as revision = 0.
+     */
+    uint32_t rev = FCFG->FCFG1_REVISION;
+    if (rev == 0xFFFFFFFF) {
+        rev = 0;
+    }
+
+    /* Enable standby in flash bank */
+    FLASH->CFG &= ~FLASH_CFG_DIS_STANDBY;
+
+    /* Select correct CACHE mode and set correct CACHE configuration */
+    rom_setup_set_cache_mode_according_to_ccfg_setting();
+
+    _trim_device(rev);
+
+    /* Set VIMS power domain control.
+     * PDCTL1VIMS = 0 ==> VIMS power domain is only powered when CPU power
+     * domain is powered
+     */
+    PRCM->PDCTL1VIMS = 0;
+
+    /* Configure optimal wait time for flash FSM in cases where flash pump
+     * wakes up from sleep */
+    FLASH->FPAC1 = (FLASH->FPAC1 & ~FLASH_FPAC1_PSLEEPTDIS_m) |
+                   (0x139 << FLASH_FPAC1_PSLEEPTDIS_s);
```

Comment:
Where does the `0x139` come from?

Reply:
It's a value provided by TI. Will add a definition to remove the magic number.